openedx/core/djangoapps/oauth_dispatch/management/commands/edx_clear_expired_tokens.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import unicode_literals
+
+import logging
+from datetime import timedelta
+
+from django.core.exceptions import ImproperlyConfigured
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.utils import timezone
+from oauth2_provider.models import AccessToken, Grant, RefreshToken
+from oauth2_provider.settings import oauth2_settings
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Clear expired access tokens and refresh tokens for Django OAuth Toolkit"
+
+    def add_arguments(self, parser):
+        parser.add_argument('--batch_size',
+                            action='store',
+                            dest='batch_size',
+                            type=int,
+                            default=1000,
+                            help='Maximum number of database rows to delete per query. '
+                                 'This helps avoid locking the database when deleting large amounts of data.')
+
+    def clear_table_data(self, query_set, batch_size, model):
+        message = 'Cleaning {} rows from {} table'.format(query_set.count(), model.__name__)
+        logger.info(message)
+        while query_set.exists():
+            qs = query_set[:batch_size]
+            batch_ids = qs.values_list('id', flat=True)
+            with transaction.atomic():
+                model.objects.filter(pk__in=list(batch_ids)).delete()
+
```

Comment:
The wait logic is missing. You need to wait a second between batches, as was done with the other management command.